spec/duration.html

```diff
@@ -1292,12 +1667,7 @@ <h1>
           1. Set _nanoseconds_ to _nanoseconds_ modulo 1000.
         1. Else,
           1. Assert: _largestUnit_ is *"nanosecond"*.
-        1. For each value _v_ of « _days_, _hours_, _minutes_, _seconds_, _milliseconds_, _microseconds_, _nanoseconds_ », do
-          1. If 𝔽(_v_) is not finite, then
-            1. If _sign_ = 1, then
-              1. Return ~positive overflow~.
-            1. Else if _sign_ = -1, then
-              1. Return ~negative overflow~.
+        1. NOTE: When _largestUnit_ is *"millisecond"*, *"microsecond"*, or *"nanosecond"*, _milliseconds_, _microseconds_, or _nanoseconds_ may be an unsafe integer. In this case, care must be taken when implementing the calculation using floating point arithmetic. It can be implemented in C++ using `std::fma()`. String manipulation will also give an exact result, since the multiplication is by a power of 10.
```

Comment:
I've thought about this case and I think this behaviour is OK. It follows the design principle of calculations using the normalized form, and at the end of the calculation converting back to duration records with ℝ(𝔽(integer)).

I see how the consequence could be surprising, but if you did `Temporal.Duration.from({nanoseconds: 9007199254740991999999999})` you'd also get a RangeError.

I'll add a test to test262 for this.

Reply:
> I see how the consequence could be surprising, but if you did `Temporal.Duration.from({nanoseconds: 9007199254740991999999999})` you'd also get a RangeError.

The problem is that `BalanceTimeDuration` is currently defined as non-throwing. 

1. `BalanceTimeDuration` is called with `norm = {[[TotalNanoseconds]]: 9007199254740991999999999}` and `largestUnit = "nanoseconds"`.
2. `BalanceTimeDuration` then calls `CreateTimeDurationRecord` with `nanoseconds = 9007199254740991999999999`. 
3. `CreateTimeDurationRecord` calls `IsValidDuration` with `nanoseconds = 9007199254740991999999999`. This call succeeds and returns `true`, because all computations are performed with mathematical integer values.
4. `CreateTimeDurationRecord` then creates a Time Duration record where the `[[Nanoseconds]]` field is `ℝ(𝔽(9007199254740991999999999)) = 9007199254740992000000000`.
5. `9007199254740992000000000` is out-side the valid range for duration nanoseconds.
6. Callers of `BalanceTimeDuration` expect that all returned duration fields are within the duration limits.

The polyfill has [this line](https://github.com/tc39/proposal-temporal/blob/133cddc4f2721d1c60269cc8e4c82192a62ff2aa/polyfill/lib/ecmascript.mjs#L3425), which isn't part of the `BalanceTimeDuration` spec.